main/src/main/scala/sbt/Extracted.scala

```diff
@@ -116,15 +115,15 @@ final case class Extracted(
   private def resolve[K <: Scoped.ScopingSetting[K] & Scoped](key: K): K =
     Scope.resolveScope(GlobalScope, currentRef.build, rootProject)(key.scope) / key
 
-  private def getOrError[T](scope: Scope, key: AttributeKey[?], value: Option[T])(implicit
+  private def getOrError[T](key: ScopedKey[?], value: Option[T])(implicit
       display: Show[ScopedKey[?]]
   ): T =
-    value getOrElse sys.error(display.show(ScopedKey(scope, key)) + " is undefined.")
+    value.getOrElse(sys.error(display.show(key) + " is undefined."))
```

Comment:
nit: While we are at it, could you check if it would be helpful to turning this into a pattern match, and inlining the method?

Reply:
I don't see anything about `Extracted` in the profiling. So probably it is not even called during loading.

But in general, yes it is good idea. I'll experiment with a `inline def` version of `getOrElse` which does pattern matching to see if it makes a difference.